localstack/services/kinesis/kinesis_listener.py

```diff
@@ -26,24 +27,29 @@ class ProxyListenerKinesis(ProxyListener):
 
     def forward_request(self, method, path, data, headers):
         global STREAM_CONSUMERS
-        data = self.decode_content(data or '{}')
+        data, encoding_type = self.decode_content(data or '{}', True)
         action = headers.get('X-Amz-Target', '').split('.')[-1]
-
         if action == 'RegisterStreamConsumer':
             prev_consumer = find_consumer(data.get('ConsumerARN', ''),
                 data.get('ConsumerName', ''), data.get('StreamARN', ''))
 
             if prev_consumer:
                 msg = 'Consumer %s already exists' % prev_consumer.get('ConsumerARN')
-                return simple_error_response(msg, 400, 'ResourceAlreadyExists')
+                return simple_error_response(msg, 400, 'ResourceAlreadyExists', encoding_type)
 
             consumer = clone(data)
             consumer['ConsumerStatus'] = 'ACTIVE'
             consumer['ConsumerARN'] = '%s/consumer/%s' % (data['StreamARN'], data['ConsumerName'])
             consumer['ConsumerCreationTimestamp'] = float(now_utc())
             consumer = json_safe(consumer)
             STREAM_CONSUMERS.append(consumer)
-            return {'Consumer': consumer}
+
+            if encoding_type is not APPLICATION_JSON:
+                consumer['ConsumerCreationTimestamp'] = int(consumer['ConsumerCreationTimestamp'])
```

Comment:
nit: wondering if we should do this cast to `int` also for JSON encoded messages? Note that `now_utc()` should already return an integer, so we could simply change line 43 to:

```
consumer['ConsumerCreationTimestamp'] = int(now_utc())
```
Would that work?